python/ray/rllib/models/catalog.py

```diff
@@ -405,19 +405,31 @@ def __init__(self, model_cls, obs_space, act_space, num_outputs, options):
         self.action_space = act_space
         self.num_outputs = num_outputs
         self.options = options
-        tf.keras.layers.Layer.__init__(self)
+        super(_KerasModelWrapper, self).__init__()
 
     def build(self, input_shape):
-        def f(input_dict):
-            return self.model_cls(input_dict, self.observation_space,
-                                  self.action_space, self.num_outputs,
-                                  self.options).outputs
+        self.template_fn = tf.make_template("keras_model_wrapper", self._make_template_fn,
+                create_scope_now_=True, custom_getter_=self._variable_getter)
+        super(_KerasModelWrapper, self).build(input_shape)
 
-        self.rllib_model = tf.keras.layers.Lambda(f)
-        tf.keras.layers.Layer.build(input_shape)
-
-    def call(self, input_dict):
-        return self.rllib_model(input_dict)
+    def call(self, input_tensor):
+        return self.template_fn(input_tensor)
 
     def compute_output_shape(self, input_shape):
         return (input_shape[0], self.num_outputs)
+
+    def _make_template_fn(self, input_tensor):
+        return self.model_cls({'obs': input_tensor}, self.observation_space,
```

Comment:
Keras layers seem to require that a tensor is passed in thus the input dict can't be passed in directly. This is because keras tries to add passed layer information onto the tensor which blows up when a dict is passed in. Not sure how this should be dealt with. This works for SAC which seems to be the only module using this class.